src/main/java/com/conveyal/r5/analyst/ExtractingGridStatisticComputer.java

```diff
@@ -0,0 +1,67 @@
+package com.conveyal.r5.analyst;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.S3Object;
+import com.google.common.io.LittleEndianDataInputStream;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.zip.GZIPInputStream;
+
+/**
+ * Extract the nth sample from a grid. Mostly useful when the grid contains a distribution rather than samples.
+ */
+public class ExtractingGridStatisticComputer {
```

Comment:
There are several other classes that have similar functions in that they compute single value grids, but they have different method signatures (cf. . We could make a superinterface and then put the extra arguments in a constructor, but that seems needlessly dogmatic for something that can easily be done with a static class. It computes a whole grid, not a single value, which is why I termed it a GridComputer and not a SampleExtractor. It seems like that would be just semantics because if we then renamed the AndrewOwenMeanGridStatisticComputer an AndrewOwenGridSampleExtractor we'd have the same problem.